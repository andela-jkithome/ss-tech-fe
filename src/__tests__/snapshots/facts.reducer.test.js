import reducer from '../../reducers/facts';

const initialState = {
  fetching: false,
  fact: {},
  error: null,
};

const fetchedFact = {
  category: ['science'],
  icon_url: 'https://assets.chucknorris.host/img/avatar/chuck-norris.png',
  id: 'uojt-t8as5ws5h1q-sjslw',
  url: 'https://api.chucknorris.io/jokes/uojt-t8as5ws5h1q-sjslw',
  value: `Nothing can escape the gravity of a black hole, except for Chuck Norris. 
  Chuck Norris eats black holes. They taste like chicken.`,
};

describe('facts reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle FACT_REQUEST', () => {
    expect(
      reducer(initialState, {
        type: 'FACT_REQUEST',
      }),
    ).toMatchSnapshot();
  });

  it('should handle FACT_SUCCESS', () => {
    expect(
      reducer(Object.assign({}, initialState, { fetching: true }), {
        type: 'FACT_SUCCESS',
        fact: fetchedFact,
      }),
    ).toMatchSnapshot();
  });

  it('should handle FACT_FAILURE', () => {
    expect(
      reducer(initialState, {
        type: 'FACT_FAILURE',
        failure: 'Error Fetching Fact!',
      }),
    ).toMatchSnapshot();
  });
});
