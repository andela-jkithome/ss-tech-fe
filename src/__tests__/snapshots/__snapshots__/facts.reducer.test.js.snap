// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`facts reducer should handle FACT_FAILURE 1`] = `
Object {
  "error": "Error Fetching Fact!",
  "fact": Object {},
  "fetching": false,
}
`;

exports[`facts reducer should handle FACT_REQUEST 1`] = `
Object {
  "error": null,
  "fact": Object {},
  "fetching": true,
}
`;

exports[`facts reducer should handle FACT_SUCCESS 1`] = `
Object {
  "error": null,
  "fact": Object {
    "category": Array [
      "science",
    ],
    "icon_url": "https://assets.chucknorris.host/img/avatar/chuck-norris.png",
    "id": "uojt-t8as5ws5h1q-sjslw",
    "url": "https://api.chucknorris.io/jokes/uojt-t8as5ws5h1q-sjslw",
    "value": "Nothing can escape the gravity of a black hole, except for Chuck Norris. 
  Chuck Norris eats black holes. They taste like chicken.",
  },
  "fetching": false,
}
`;

exports[`facts reducer should return the initial state 1`] = `
Object {
  "error": null,
  "fact": Object {},
  "fetching": false,
}
`;
